### DISCUSSION TITLE
Better Flask 1.x detection

### DISCUSSION THREAD

**Discussion by FrankyBoy** [maxcountryman/flask-login#662] [Ideas]
Hi,

Right now you are using try/catch as a flask 1.x fallback: https://github.com/maxcountryman/flask-login/blob/main/src/flask_login/utils.py#L276

This is problematic, because it messes with log messages. I.e. if my application throws an exception, I would not see [my exception], but "Attribute Error [...] During handling of the exception another exception was raised: [my exception]", which clutters the log with useless information and makes the actually relevant infos harder to spot.

I saw a better way in the flask-jwt-edtended docs: https://flask-jwt-extended.readthedocs.io/en/stable/_modules/flask_jwt_extended/view_decorators/

```
# Compatibility with flask < 2.0
if hasattr(current_app, "ensure_sync") and callable(getattr(current_app, "ensure_sync", None)):
    return current_app.ensure_sync(fn)(*args, **kwargs)

return fn(*args, **kwargs)
```

Would be cool if you could use this method.

Cheers :)

**Comment by PythonCoderAS** [comment]
You could simplify this down:

```py
# Compatibility with flask < 2.0
if callable(getattr(current_app, "ensure_sync", None)):
    return current_app.ensure_sync(fn)(*args, **kwargs)

return fn(*args, **kwargs)
```

Since None isn't callable it'll fail anyways.

**Reply by jonasps** [reply]
Looks good to me, create a PR?

**Reply by FrankyBoy** [reply]
There ya go ... https://github.com/maxcountryman/flask-login/pull/667 ... can I say editing stuff online in github is a chore compared to gitlab with its nice web-ide ;)
